test/edu/ucsd/library/xdre/tab/BatchEditExcelSourceTest.java

```diff
@@ -157,5 +158,17 @@ public void testNextRecord() throws Exception {
 
         // collection(s)
         assertEquals("collection(s) doesn't match!", "Collection(s)", data.get("collection(s)"));
+
+        // Component
+        List<TabularRecord> comps = editRecord.getComponents();
+        assertEquals("component size doesn't match!", 1, comps.size());
+        Map<String, String> compData = comps.get(0).getData();
+        assertEquals("component title doesn't match!", "Test Component 1", compData.get("title"));
+
+        // Sub-Component

```

Comment:
This is entirely up to you, but a question I have is should we have some kind of test/validation that ensures no one tries to accidentally(or intentionally..) do a sub-sub Component? Basically test/verify that we're only supporting a sub-component level of depth? I think you would know better than me if that's a legitimate concern for your implementation. Just a question that popped into my mind while reviewing these tests.

Reply:
Good question. I don't think it'll work with deeper level of depth with the logic and implementation we have for the Excel InputStream: rows of `Component` and `Sub-component`. It'll be always a new Component OR a Sub-component that under the current component context.

Reply:
Ok sounds good. Thanks.